### PR TITLE
Add deletion of pending regional organizations

### PR DESCRIPTION
**Pull request by saranshgrover**
Fixes part of #5119 

Still need to add tests, where I was struggling a little. Would appreciate some guidance with that (and a general review)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -32,3 +32,11 @@ webroot/forum/files/*
 webroot/forum/images/avatars/gallery/*
 webroot/forum/images/avatars/upload/*
 webroot/forum/store/*
+
```
Comment (gregorbg) [inline .gitignore:35]: This entire section can be outsourced to your personal (i.e. local) gitignore as designated by Philippe: https://github.com/thewca/worldcubeassociation.org/pull/6312

File: WcaOnRails/app/controllers/regional_organizations_controller.rb [hunk 2]
```diff
@@ -31,6 +31,25 @@ def update
     end
   end
 
+  def destroy
+    @regional_organization = regional_organization_from_params
+    if current_user.can_manage_regional_organizations?
```
Comment (gregorbg) [inline WcaOnRails/app/controllers/regional_organizations_controller.rb:36]: Why do you check for permissions in the destroy method, whey you could just as easily not exclude it from the general permissions check at line 5?

File: WcaOnRails/app/controllers/regional_organizations_controller.rb [hunk 3]
```diff
@@ -31,6 +31,25 @@ def update
...
+      if @regional_organization.start_date.nil?
```
Comment (gregorbg) [inline WcaOnRails/app/controllers/regional_organizations_controller.rb:37]: Add a separate `is_pending?` method to the RO entity which abstracts away this change. Can also be reused in the HTML view.

File: WcaOnRails/app/views/regional_organizations/_regional_organizations_table.html.erb [hunk 4]
```diff
@@ -18,6 +18,19 @@
             <%= ui_icon("edit") %> Manage
           <% end %>
         </td>
+        <td>
+        <% if ro.start_date == nil %>
```
Comment (gregorbg) [inline WcaOnRails/app/views/regional_organizations/_regional_organizations_table.html.erb:22]: See remark about adding and reusing `is_pending?` above.

File: WcaOnRails/config/locales/en.yml [hunk 5]
```diff
@@ -1993,6 +1993,8 @@ en:
       end_date_after_start_date: "cannot be after Acknowledged until"
     create:
       create_success: "Application successfully submitted! Please check your email, and wait for the WCA Board of Directors to review it"
+    delete:
+      confirm_delete: "Are you sure you want to delete this Regional Organization?"
```
Comment (gregorbg) [inline WcaOnRails/config/locales/en.yml:1997]: Maybe nit, but "delete this Regional Organization **request**"?
I mean, you're not destroying the organization itself I hope :joy:

**Review by gregorbg — CHANGES_REQUESTED** [review]
A few nitpicks and one actual concern about reusing the "is RO pending?" logic.

File: .gitignore (hunk 1 above)
Comment (saranshgrover) [reply .gitignore:35]: Woah - had no idea this was a thing!

**Review by gregorbg — APPROVED** [review]
Thanks, LGTM!

**Comment by saranshgrover** [comment]
Can this be merged?

File: WcaOnRails/spec/requests/regional_organizations_spec.rb [hunk 6]
```diff
@@ -134,6 +134,7 @@
     context "when signed in as a board member and with invalid params" do
       it "renders the new regional organization form" do
         sign_in board_member
+        post new_regional_organization_path, params: { regional_organization: valid_attributes }
```
Comment (gregorbg) [inline WcaOnRails/spec/requests/regional_organizations_spec.rb:137]: What is this change doing? The context is called `context "when signed in as a board member and with invalid params" do` which seems to suggest that we should *not* post `valid_attributes` here.

**Review by gregorbg — COMMENTED** [review]
Unsure about one of the changes in the rspec tests. Everything else LGTM!

File: WcaOnRails/spec/requests/regional_organizations_spec.rb (hunk 6 above)
Comment (saranshgrover) [reply WcaOnRails/spec/requests/regional_organizations_spec.rb]: whoops - I think I accidentally added it, removed it now. Thanks for spotting!

**Pull request merged**